editors/code/src/highlighting.ts

```diff
@@ -32,12 +69,25 @@ export class Highlighter {
             tag: string,
             textDecoration?: string
         ): [string, vscode.TextEditorDecorationType] => {
-            const color = new vscode.ThemeColor('ralsp.' + tag);
-            const decor = vscode.window.createTextEditorDecorationType({
-                color,
-                textDecoration
-            });
-            return [tag, decor];
+            const rule = scopesMapper.toRule(tag, scopes.find);
+
+            if (rule) {
+                const decor = createDecorationFromTextmate(rule);
+                return [tag, decor];
+            } else {
+                const fallBackTag = 'ralsp.' + tag;
+                // console.log(' ');
```

Comment:
I think we can use a channel for logging (see `window.createOutputChannel`).

Reply:
@lnicola Can you give me an example on how it's used and where it ends up? I need a place to point out which parts aren't highlighted so the user can add them.

Reply:
You just call it somewhere and then use `appendLine` on the result to add a line. Look at https://code.visualstudio.com/api/references/vscode-api#window or the two places it's used in `rust-analyzer`. Unfortunately, I think it will always create a new channel instead of returning an existing one, so you will want to store the reference somewhere. It's probably fine to use a new channel for the theme stuff.